fabric/src/main/java/com/discordsrv/fabric/module/chat/FabricAdvancementModule.java

```diff
@@ -44,25 +42,49 @@ public FabricAdvancementModule(FabricDiscordSRV discordSRV) {
+    /*public static void onGrant(Advancement advancement, ServerPlayerEntity owner) {
+    *///?} else {
+    public static void onGrant(net.minecraft.advancement.AdvancementEntry advancementEntry, ServerPlayerEntity owner) {
+    //?}
         if (instance == null || !instance.enabled) return;
 
         FabricDiscordSRV discordSRV = instance.discordSRV;
+        //? if minecraft: <1.20.2 {
+        /*AdvancementDisplay display = advancement.getDisplay();
+         *///?} else {
         Advancement advancement = advancementEntry.value();
+        AdvancementDisplay display = advancement.display().orElse(null);
+        //?}
 
-        Optional<AdvancementDisplay> displayOptional = advancement.display();
-        if (displayOptional.isEmpty() || !displayOptional.get().shouldAnnounceToChat()) {
-            instance.logger().trace("Skipping advancement display of \"" + (advancement.name().isPresent() ? advancement.name().get() : advancement) + "\" for "
+        if (display == null || !display.shouldAnnounceToChat()) {
+            instance.logger().trace("Skipping advancement display of \"" + (advancement) + "\" for "
                     + owner + ": advancement display == null or does not broadcast to chat");
             return;
         }
 
-        AdvancementDisplay display = displayOptional.get();
         AdvancementFrame frame = display.getFrame();
 
-        MinecraftComponent message = ComponentUtil.toAPI(discordSRV.getAdventure().asAdventure(frame.getChatAnnouncementText(advancementEntry, owner)));
-        MinecraftComponent title = ComponentUtil.toAPI(discordSRV.getAdventure().asAdventure(display.getTitle()));
-        MinecraftComponent description = ComponentUtil.toAPI(discordSRV.getAdventure().asAdventure(display.getDescription()));
+        //? if minecraft: <1.20.3 {
+        /*Text rawChat = Text.translatable("chat.type.advancement." + frame.getId(), owner.getDisplayName(), display.getTitle());
+        *///?} else {
+        Text rawChat = frame.getChatAnnouncementText(advancementEntry, owner);
+        //?}
+        Text rawTitle = display.getTitle();
+        Text rawDesc  = display.getDescription();
+
+        //? if adventure: <6 {
+        /*@SuppressWarnings("removal")
+        MinecraftComponent message = ComponentUtil.toAPI(discordSRV.getAdventure().toAdventure(rawChat));
+        @SuppressWarnings("removal")
+        MinecraftComponent title = ComponentUtil.toAPI(discordSRV.getAdventure().toAdventure(rawTitle));
+        @SuppressWarnings("removal")
+        MinecraftComponent description = ComponentUtil.toAPI(discordSRV.getAdventure().toAdventure(rawDesc));
+        *///?} else {
+        MinecraftComponent message = ComponentUtil.toAPI(discordSRV.getAdventure().asAdventure(rawChat));
+        MinecraftComponent title = ComponentUtil.toAPI(discordSRV.getAdventure().asAdventure(rawTitle));
+        MinecraftComponent description = ComponentUtil.toAPI(discordSRV.getAdventure().asAdventure(rawDesc));
+        //?}
```

Comment:
Moving the Text -> Component (and reverse) conversion logic to a shared method (in a util class, ComponentFactory or even FabricDiscordSRV) would be way more maintainable, accounting for future changes